### ISSUE TITLE
Different behavior with splitting on/off and a function with removable singularity

### ISSUE THREAD

**Issue by gradywright** [chebfun/chebfun#1615]
Consider the function `g = @(x) log((x+3)/4)./(x-1)`, which comes from ATAP exercise 8.4 (c).  This function has a removable singularity at `x=1` (and branch point at `x=-3`).  Constructing `g` with standard Chebfun input gives:
```
f = chebfun(g)
Warning: Function not resolved using 65537 pts. Have you tried 'splitting on'? 
> In chebfun.constructor>constructorNoSplit at 120
  In chebfun.constructor at 63
  In chebfun.chebfun>chebfun.chebfun at 219 
f =
   chebfun column (1 smooth piece)
       interval       length     endpoint values  
[      -1,       1]    65537      0.35     0.25 
vscale = 3.465736e-01.
```
Whereas constructing with `splitting on` gives
```
fso = chebfun(g,'splitting','on')
fso =
   chebfun column (1 smooth piece)
       interval       length     endpoint values  
[      -1,       1]       18      0.35     0.25 
vscale = 3.465736e-01.
```
Plotting the coefficients for the former case may help to illustrate what happens.

Note that the accuracy of the splitting on case is about twice as high as the splitting off (unresolved) case:
```
>> x = linspace(-1,1,1001)'; x = x(2:end-1);
>> norm(g(x)-fso(x),'inf')
ans =
     9.225953334635051e-14
>> norm(g(x)-f(x),'inf')
ans =
     4.113376306236205e-14
```

**Comment by aaustin141** [comment]
This may be a machine-dependent thing, as the constructor without splitting works just fine for me:
```
>> g = @(x) log((x + 3)/4)./(x - 1);
>> f = chebfun(g)
f =
   chebfun column (1 smooth piece)
       interval       length     endpoint values  
[      -1,       1]       19      0.35     0.25 
vscale = 3.465736e-01.
```

**Comment by gradywright** [comment]
I just updated to the development branch and now it works fine.  Perhaps this is because of the changes to `sampleTest`?

**Comment by aaustin141** [comment]
Yep, that's probably it.  I just tried on fd604a18 (the commit prior to the merge of the `sampleTest` changes), and it failed for me, too.  Since the removable singularity in your example is at the right endpoint, my guess is that it's the removal of the code that did the sample test at some points near the endpoints that fixed things.

**Comment by gradywright** [comment]
Fixed in #1614
